pkg/controller/networkpolicy/reconciler.go

```diff
@@ -474,12 +516,13 @@ func (r *Reconciler) reconcileNetworkPolicyAllowToDNS(ctx context.Context, log l
 					// required for node local dns feature, allows egress traffic to CoreDNS
 					{
 						IPBlock: &networkingv1.IPBlock{
-							CIDR: fmt.Sprintf("%s/32", runtimeDNSServerAddress),
+							CIDR: fmt.Sprintf("%s/%d", runtimeDNSServerAddress, runtimeDNSServerAddressBitLen),
 						},
 					},
 					// required for node local dns feature, allows egress traffic to node local dns cache
 					{
 						IPBlock: &networkingv1.IPBlock{
+							// node local dns feature is only supported for shoots with IPv4 single-stack networking
 							CIDR: fmt.Sprintf("%s/32", nodelocaldnsconstants.IPVSAddress),
```

Comment:
No technical reason, we considered it out-of-scope for this PR. Is this something that would be required?

Reply:
`node-local-dns` is not enabled per default in the local setup. Therefore, you can indeed argue that it does not need to be included in this pull request.
Nevertheless, I would still like to have this tracked somewhere. This could be an additional point in https://github.com/gardener/gardener/issues/7051 or a separate issue. Otherwise, we will most likely forget it.

WDYT?

Reply:
100% Agree. Let's track it!

Reply:
Will you add it to https://github.com/gardener/gardener/issues/7051 or do you prefer to track it in a separate issue?

Reply:
Added it to #7051